tests/source/float-lit-trailing-zero-always.rs

```diff
@@ -0,0 +1,17 @@
+// rustfmt-float_literal_trailing_zero: Always
+
+fn float_literals() {
+    let a = 0.;
+    let b = 0.0;
+    let c = 100.;
+    let d = 100.0;
+    let e = 5e3;
+    let f = 5.0e3;
+    let g = 7f32;
+    let h = 7.0f32;
+    let i = 9e3f32;
+    let j = 9.0e3f32;
+    let k = 1000.00;
+    let l = 1_000_.;
+    let m = 1_000_.000_000;
+}
```

Comment:
This is a good start, but I'd really like to see us expand on these test cases.

1. Can we move all `float_literal_trailing_zero` tests to a `tests/{source|target}/config/float_literal_trailing_zero` directory. Then you can name each test file as the name of the enum variant.
2. To cover all our bases let's add a test case for `Preserve`. We should only need a `target` test file for the preserve case since we don't expect formatting to change.
3. Let's try to be as thorough as we can when coming up with test cases. You've already highlighted let bindings, but there are plenty of other places where float literals can be written. A few that come to mind are conditional expressions, struct literals, match patterns, function arguments, array literals, macro calls, etc.
 
One thing that I'm particularly interested in seeing is what happens when we're at or near the `max_width` boundary. Does adding the extra `0` correctly wrap a function call or array literal when adding the extra `0` pushes us over the `max_width`, and do we correctly format on a single line when adding `0` pushes us to exactly the `max_width`.

Reply:
> Also please tell me if I did the tests right. Three separate test files seem like a lot for such a small feature, but I don't see how else I could check all configurations.

Oh and to answer your question, this is correct. You'll need to create individual test files for each case!

Reply:
1. Done.
2. Done.
3. I've added one macro and one non-macro example that causes a line wrapping change. Do you think we really need to cover all these cases (conditional expressions, struct literals, match patterns, function arguments, etc.)? To me it feel a bit excessive. And it seems different from the current testing strategy (take `hex_literal_case` tests as a close example).